tracing-opentelemetry/src/layer.rs

```diff
@@ -253,6 +254,27 @@ impl<'a> field::Visit for SpanAttributeVisitor<'a> {
             _ => self.record(Key::new(field.name()).string(format!("{:?}", value))),
         }
     }
+
+    /// Set attributes on the underlying OpenTelemetry [`Span`] from values that
+    /// implement Debug. Also adds the `source` chain as an extra field
+    ///
+    /// [`Span`]: opentelemetry::trace::Span
+    fn record_error(
+        &mut self,
+        field: &tracing_core::Field,
+        value: &(dyn std::error::Error + 'static),
+    ) {
+        let mut chain = Vec::new();
+        let mut next_err = value.source();
+
+        while let Some(err) = next_err {
+            chain.push(Cow::Owned(format!("{}", err)));
+            next_err = err.source();
+        }
+
+        self.record(Key::new(field.name()).string(format!("{}", value)));
```

Comment:
nit, take it or leave it: it may be somewhat nicer to use `to_string()` instead of `format!` for formatting these using `Display`?
```suggestion
            chain.push(Cow::Owned(err.to_string()));
            next_err = err.source();
        }

        self.record(Key::new(field.name()).string(value.to_string()));
```